blocks/motion-background/motion-background.js

```diff
@@ -0,0 +1,351 @@
+		}
+		return;
+	}
+
+	const editor = document.getElementById( 'editor' );
+	if ( editor ) {
+		editor.parentNode.insertBefore( gl.canvas, editor );
+	} else {
+		document.body.insertBefore( gl.canvas, document.body.firstChild );
+	}
+
+	const vertexShader = `
+		attribute vec3 position;
+		attribute vec2 texcoord;
+
+		varying vec2 uv;
+	
+		void main () {
+			uv = texcoord;
+			gl_Position = vec4( position, 1. );
+		}
+	`;
+
+	const fragmentShader = `
+		precision mediump float;
+
+		#define SRGB_TO_LINEAR( c ) pow( c, vec3( 2.2 ) )
+		#define LINEAR_TO_SRGB( c ) pow( c, vec3( 1.0 / 2.2 ) )
+
+		uniform vec3 color1;
+		uniform vec3 color2;
+		uniform vec3 color3;
+		uniform vec3 color4;
+
+		varying vec2 uv;
+
+		void main()
+		{      
+			vec3 tl = SRGB_TO_LINEAR( color1 );
+			vec3 tr = SRGB_TO_LINEAR( color2 );
+			vec3 bl = SRGB_TO_LINEAR( color3 );
+			vec3 br = SRGB_TO_LINEAR( color4 );
+
+			vec3 top = mix( tl, tr, uv.x );
+			vec3 bottom = mix( bl, br, uv.x );
+			vec3 gradient =  mix( bottom, top, uv.y ) ;
+
+			gl_FragColor = vec4( LINEAR_TO_SRGB( gradient ), 1. );
+		}
+	`;
+
+	const vertexShaderEffect = `
+		attribute vec3 position;
+		attribute vec2 texcoord;
+
+		uniform vec2 resolution;
+
+		varying vec2 uv;
+	
+		void main () {
+			uv = texcoord * normalize( resolution );
+			gl_Position = vec4( position, 1. );
+		}
+	`;
+
+	const fragmentShaderEffect = `
+		precision mediump float;
+
+		#define MAX_COMPLEXITY 32
+		#define MIRRORED_REPEAT( p ) abs( 2. * fract( p / 2. ) - 1. )
+
+		uniform float time;
+
+		uniform vec2 mouse;
+		uniform int complexity;
+		uniform float mouseSpeed;
+		uniform float fluidSpeed;
+		uniform sampler2D texture;
+
+		varying vec2 uv;
+
+		void main() {
+			vec2 c = uv;
+			for ( int i = 1; i < MAX_COMPLEXITY; i++ ) {
+				if ( i >= complexity ) continue;
+				float f = float( i );
+				c += ( time * 0.001 );
+				c.x += 0.6 / f * sin( ( f * c.y ) + ( time / fluidSpeed ) + ( 0.3 * f ) );
+				c.x += 0.5 + ( mouse.y / mouseSpeed );
+				c.y += 0.6 / f * sin( ( f * c.x ) + ( time / fluidSpeed ) + ( 0.3 * f + 10. ) );
+				c.y -= 0.5 - ( mouse.x / mouseSpeed );
+			}
+			gl_FragColor = texture2D( texture, MIRRORED_REPEAT( c ) );
+		}
+	`;
+
+	const programInfoGradient = twgl.createProgramInfo( gl, [ vertexShader, fragmentShader ] );
+	const programInfoEffectPass = twgl.createProgramInfo( gl, [ vertexShaderEffect, fragmentShaderEffect ] );
+
+	const screenBufferInfo = twgl.createBufferInfoFromArrays( gl, {
+		position: [ -1, -1, 0, 1, -1, 0, -1, 1, 0, -1, 1, 0, 1, -1, 0, 1, 1, 0 ], // vec3
+		texcoord: [ 0, 0, 1, 0, 0, 1, 0, 1, 1, 0, 1, 1 ], // vec2
+	} );
+
+	const textureInfos = new WeakMap();
+
+	const globalState = {
+		time: window.performance.now(),
+		mouse: [ 0, 0 ],
+	};
+
+	/**
+	 * Hack to move the background behind the canvas since the canvas is added
+	 * underneath the editor to allow for UI elements to draw overtop.
+	 */
+	function replaceBackground() {
```

Comment:
@jasmussen This is one of the "hacky things" I had to do to get layering to look good.

In order to allow for more than 2-32 blocks (depending on the device/browser/heuristics they use for the maximum number of webgl contexts), I have a [single canvas for all the blocks](https://webglfundamentals.org/webgl/lessons/webgl-multiple-views.html). In order for text and UI elements to draw on top of it, the canvas is placed underneath the `#editor` div. However, themes like Twentytwenty have a background color applied to the `.editor-styles-wrapper` div which is rendered by React. Since React owns it and I don't have access to React from here, this is the best I could think of.

One possibility if we want to change the editor might be to have a place for things like this in the React DOM that components can render into with a [portal](https://reactjs.org/docs/portals.html); although, I don't know how practical that would be.

Reply:
Ah, I suspect this is what's causing the gray background I just mentioned. That was in TwentyTwenty.

I've no problem with hacks per se, but we'll probably want to find a workaround here, as `editor-styles-wrapper` is the equivalent of the document root for the theme. It is a class that exists to attach editor styles to, so it is expected for themes that style the editor to put background colors and other things here. [Here's more](https://developer.wordpress.org/block-editor/developers/themes/theme-support/#editor-styles), but the gist is that if you load `editor-style.scss` in using `add_editor_style`, the following:

```
body {
background: red;
}
```

gets _rewritten_ / transcoded into the following:

```
.editor-styles-wrapper {
background: red;
}
```

Is there an upstream patch we can consider to enable webGL blocks like this?